Collapse tracing_state.h into tracer.h

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21515 Add WeakIValue, use in tracer.
* #21514 clean up the TracingState API
* **#21513 Collapse tracing_state.h into tracer.h**
* #21325 Interpreter support for CallFunction/CallMethod
* #21273 Expose ExecutionPlan in prep for function calls
* #21272 Add flag to temporarily enable first class modules
* #21240 Reduce number of stack manipulation instructions in interpreter.
* #21185 Prepare interpreter for function calling

Summary: Whatever reasons for the separation originally, almost
everywhere just includes tracer.h now, making the distinction confusing.
This just collapses them to a single file.

Differential Revision: [D15719982](https://our.internmc.facebook.com/intern/diff/D15719982)

Commit:
Collapse tracing_state.h into tracer.h

Summary: Whatever reasons for the separation originally, almost
everywhere just includes tracer.h now, making the distinction confusing.
This just collapses them to a single file.

suo reviewed (approved):
accepting assuming the changes are mechanical

Commit:
Update on "Collapse tracing_state.h into tracer.h"

Collapse tracing_state.h into tracer.h

Summary: Whatever reasons for the separation originally, almost
everywhere just includes tracer.h now, making the distinction confusing.
This just collapses them to a single file.

gh-metadata: pytorch pytorch 21513 gh/zdevito/49/head

zdevito committed:
Update on "Collapse tracing_state.h into tracer.h"

Collapse tracing_state.h into tracer.h

Summary: Whatever reasons for the separation originally, almost
everywhere just includes tracer.h now, making the distinction confusing.
This just collapses them to a single file.

gh-metadata: pytorch pytorch 21513 gh/zdevito/49/head

zdevito committed:
Update on "Collapse tracing_state.h into tracer.h"

Collapse tracing_state.h into tracer.h

Summary: Whatever reasons for the separation originally, almost
everywhere just includes tracer.h now, making the distinction confusing.
This just collapses them to a single file.

gh-metadata: pytorch pytorch 21513 gh/zdevito/49/head
